infrastructure/cdn-in-a-box/readiness/run.sh

```diff
@@ -30,20 +30,20 @@ while ! to-ping; do
     sleep 3
 done
 
+xmlID="$(<<<"$DS_HOSTS" sed 's/ .*//g')" # Only get the first xmlID
 while true; do
     sleep 3
-    exampleURLs=($(to-get /api/${TO_API_VERSION}/deliveryservices | jq -r '.response[].exampleURLs[]' || true))
-    if ! exampeURLs=($(to-get "/api/${TO_API_VERSION}/deliveryservices" | jq -r '.response[].exampleURLs[]')) \
-            || [[ "${#exampleURLs[@]}" -eq 0 ]]; then
-        echo waiting for delivery service example URLs
+    if ! exampleURLs=($(to-get "/api/${TO_API_VERSION}/deliveryservices?xmlId=${xmlID}" | jq -r '.response[].exampleURLs[]')) ||
```

Comment:
I think the reason you split this up into first and second xmlids was because it wasn't properly checking all example URLs for all Delivery Services, right? I think the intention was to support an arbitrary number of Delivery Services (>0) - can we just loop over the expected

Reply:
> I think the reason you split this up into first and second xmlids was because it wasn't properly checking all example URLs for all Delivery Services, right?

I split them up because the checks for the first and second DSes are different.

For the `demo1` DS, the check is to GET the example URLs, follow redirects, and check status codes.

For the `demo2` DS, the check is to verify the Federation CNAME is returned when querying the Delivery Service hostname.

> can we just loop over the xmlids to support as many as exist?

If we do that, we'd need to remove `demo3` from the `DS_HOSTS` environment variable so we can expect all DSes listed to exist:

https://github.com/apache/trafficcontrol/blob/473d9c242247618f17ac64a2b4e53e7e14fdc7ed/infrastructure/cdn-in-a-box/variables.env#L21

I suppose we could loop over the Delivery Service hosts given in DS_HOSTS and
- GET example URLs if it's an HTTP-routed Delivery Service
- Find the Federation using the `FEDERATION_CNAME` environment variable, get its Delivery Service's `xmlID` from that, and for that `xmlID`, expect the value `FEDERATION_CNAME` as the response
- `dig` the Delivery Service hostname if it's a DNS-routed Delivery Service that is not used by the Federation specified by `FEDERATION_CNAME`

Reply:
That approach would also involve exiting with an error if at least 1 Delivery Services of each type we expect to check is not found.

Reply:
Yes that sounds very convoluted - never mind